packages/cms/pages/settings/roles-and-permissions/roles-and-permissions.tsx

```diff
@@ -0,0 +1,221 @@
+    }
+
+    fetchAdminRoles()
+  }, [])
+
+  const resources = window.Tensei.state.resources
+  return (
+    <>
+      {resources.map(resource => (
+        <AccordionWrapper>
+          <AccordionHeader>{resource.name}</AccordionHeader>
+          <Accordionbody>
+            {adminRoles.map(
+              role =>
+                role.slug.split(':')[1] ===
+                  resource.namePlural.toLowerCase() && (
+                  <AccordionItem>
+                    <EuiFlexGroup justifyContent="spaceBetween">
+                      <EuiText size="s">Can {role.name}</EuiText>
+                      <Switch />
+                    </EuiFlexGroup>
+                  </AccordionItem>
+                )
+            )}
+          </Accordionbody>
+        </AccordionWrapper>
+      ))}
+    </>
+  )
+}
+const RolesTable: React.FC = () => {
+  const [items, setItems] = useState<any>([])
+  const [memberNo, setMemberNo] = useState<any>([])
+  const [loading, setLoading] = useState(true)
+  const [isFlyoutOpen, setIsFlyoutOpen] = useState(false)
+
+  const columns = [
+    {
+      field: 'Role',
+      name: 'Role '
+    },
+    {
+      field: 'Members',
+      name: 'Members'
+    }
+  ]
+
+  useEffect(() => {
+    const fetchAdminRoles = async () => {
+      const [data] = await window.Tensei.api.get('admin-roles')
+      const [result] = await window.Tensei.api.get(
+        'admin-roles?populate=adminUsers'
+      )
+      console.log(result)
+      setItems(data?.data.data)
+      setMemberNo(result?.data.data)
+      setLoading(false)
+    }
+    fetchAdminRoles()
+  }, [])
+
+  const renderedItems = items.map((item: any) => {
+    const numberOfMembers = memberNo?.filter(
+      (memberItem: any) => item.name === memberItem.name
+    )
+    const member = numberOfMembers.length > 1 ? 'members' : 'member'
+    return {
+      ...item,
+      Role: item.name,
+      Members: `${numberOfMembers.length} ${member}`
+    }
+  })
+
+  const Switch = () => {
+    const [checked, setChecked] = useState(true)
+    return (
+      <EuiSwitch
+        label="Enable"
+        checked={checked}
+        onChange={e => setChecked(!checked)}
+      />
+    )
+  }
+
+  const RolesFlyout: React.FC = () => {
+    const flyoutHeadingId = useGeneratedHtmlId()
+
+    const closeFlyout = () => {
+      setIsFlyoutOpen(false)
+    }
+
+    return (
+      <EuiFlyout onClose={closeFlyout}>
+        <EuiFlyoutHeader hasBorder aria-labelledby={flyoutHeadingId}>
+          <EuiTitle>
+            <h2 id={flyoutHeadingId}> Create custom role </h2>
+          </EuiTitle>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody>
+          <EuiText size="s">Role Name</EuiText>
+          <EuiFieldText />
+          <EuiText>Description</EuiText>
+          <EuiTextArea />
+          <EuiSpacer size="xl" />
+          <FlyoutAccordion />
+        </EuiFlyoutBody>
+        <EuiFlyoutFooter>
+          <EuiFlexGroup justifyContent="flexEnd">
```

Comment:
Let's keep the consistency with the other fly outs in the application. the buttons in the footer should be spaced apart. Also, the cancel button should be the same as we have on other pages.